torch/_dynamo/external_utils.py

```diff
@@ -17,3 +17,13 @@ def inner(*args, **kwargs):
         return fn(*args, **kwargs)
 
     return inner
+
+
+def call_hook(hook, *args):
+    """
+    Used by compiled autograd to handle hook returning None
+    """
+    result = hook(*args)
+    if result is None:
+        return args[0]
```

Comment:
what kind of hook is this handling? This should be `args` no?

Reply:
This handles:
- `post_hook(node_outputs, node_inputs)`
- `pre_hook(node_inputs)`

For the post_hook, you only return the first arg.

Reply:
What about the pre-hooks?

Reply:
For pre_hooks there will only be 1 arg, so len(args)==1 and args[0] is the only arg to the function.

```
def foo(*args):
   print(repr(args[0]))

foo(10)
```
will output `10`.  `args` will be `(10,)`